test/controllers/offer.controller.test.js

```diff
@@ -1,677 +1,815 @@
-//     it('returns the error', (done) => {
-//       sinon.stub(Offer, 'aggregate').throws(new Error('Type Error'));
-//       request()
-//         .get(`/api/v1/offer/${offerId}`)
-//         .set('authorization', userToken)
-//         .end((err, res) => {
-//           expect(res).to.have.status(500);
-//           done();
-//           Offer.aggregate.restore();
-//         });
-//     });
-//   });
-// });
+import mongoose from 'mongoose';
+import { expect, request, sinon, useDatabase } from '../config';
+import Offer from '../../server/models/offer.model';
+import User from '../../server/models/user.model';
+import OfferFactory from '../factories/offer.factory';
+import EnquiryFactory from '../factories/enquiry.factory';
+import UserFactory from '../factories/user.factory';
+import PropertyFactory from '../factories/property.factory';
+import { createOffer } from '../../server/services/offer.service';
+import { addEnquiry } from '../../server/services/enquiry.service';
+import { addUser } from '../../server/services/user.service';
+import { addProperty } from '../../server/services/property.service';
+import { OFFER_STATUS } from '../../server/helpers/constants';
+
+useDatabase();
+
+let adminToken;
+let userToken;
+const userId = mongoose.Types.ObjectId();
+const adminId = mongoose.Types.ObjectId();
+const adminUser = UserFactory.build({ _id: adminId, role: 0, activated: true });
+const regualarUser = UserFactory.build({ _id: userId, role: 1, activated: true });
+const propertyId = mongoose.Types.ObjectId();
+const property = PropertyFactory.build({ _id: propertyId, addedBy: adminId, updatedBy: adminId });
+
+describe('Offer Routes', () => {
+  beforeEach(async () => {
+    adminToken = await addUser(adminUser);
+    userToken = await addUser(regualarUser);
+    await addProperty(property);
+  });
+
+  describe('Create Offer Route', () => {
+    const enquiryId = mongoose.Types.ObjectId();
+    const enquiry = EnquiryFactory.build({ _id: enquiryId, userId: adminId, propertyId });
+
+    beforeEach(async () => {
+      await addEnquiry(enquiry);
+    });
+
+    context('with valid data', () => {
+      it('returns successful offer', (done) => {
+        const offer = OfferFactory.build({ enquiryId });
+        request()
+          .post('/api/v1/offer/create')
+          .set('authorization', adminToken)
+          .send(offer)
+          .end((err, res) => {
+            expect(res).to.have.status(201);
+            expect(res.body.success).to.be.eql(true);
+            expect(res.body.message).to.be.eql('Offer created');
+            expect(res.body.offer).to.have.property('vendorInfo');
+            expect(res.body.offer).to.have.property('enquiryInfo');
+            expect(res.body.offer).to.have.property('propertyInfo');
+            expect(res.body.offer).to.have.property('userInfo');
+            done();
+          });
+      });
+    });
+
+    context('when user token is used', () => {
```

Comment:
Since you are deleting the token, I feel it is better to explicitly state that an invalid token is used here
```suggestion
    context('when an invalid token is used', () => {
```